test/e2e/garbage_collector.go

```diff
@@ -275,4 +342,53 @@ var _ = framework.KubeDescribe("Garbage collector", func() {
 		}
 		gatherMetrics(f)
 	})
+
+	It("Should delete RS created by deployment when not orphaning", func() {
+		clientSet := f.ClientSet
+		deployClient := clientSet.Extensions().Deployments(f.Namespace.Name)
+		rsClient := clientSet.Extensions().ReplicaSets(f.Namespace.Name)
+		deploymentName := "simpletest.deployment"
+		deployment := newOwnerDeployment(f, deploymentName)
+		By("create the deployment")
+		createdDeployment, err := deployClient.Create(deployment)
+		if err != nil {
+			framework.Failf("Failed to create deployment: %v", err)
+		}
+		// wait for deployment to create some rs
+		By("Wait for the Deployment to create new ReplicaSet")
+		err = wait.PollImmediate(500*time.Millisecond, 1*time.Minute, func() (bool, error) {
+			rsList, err := rsClient.List(v1.ListOptions{})
+			if err != nil {
+				return false, fmt.Errorf("Failed to list rs: %v", err)
+			}
+			return len(rsList.Items) > 0, nil
+
+		})
+		if err == wait.ErrWaitTimeout {
+			err = fmt.Errorf("Failed to wait for the Deployment to create some ReplicaSet: %v", err)
+		}
+
+		By("delete the deployment")
+		deleteOptions := getNonOrphanOptions()
+		deleteOptions.Preconditions = v1.NewUIDPreconditions(string(createdDeployment.UID))
+		if err := deployClient.Delete(deployment.ObjectMeta.Name, deleteOptions); err != nil {
+			framework.Failf("failed to delete the deployment: %v", err)
+		}
+		By("wait for all rs to be garbage collected")
+		err = wait.PollImmediate(500*time.Millisecond, 1*time.Minute, func() (bool, error) {
+			return verifyRemainingDeploymentsAndReplicaSets(f, clientSet, deployment, 0, 0)
+		})
+		if err == wait.ErrWaitTimeout {
+			err = fmt.Errorf("Failed to wait for all rs to be garbage collected: %v", err)
+			remainingRSs, err := rsClient.List(v1.ListOptions{})
+			if err != nil {
+				framework.Failf("failed to list RSs post mortem: %v", err)
+			} else {
+				framework.Failf("remaining rs are: %#v", remainingRSs)
+			}
+
+		}
+
+		gatherMetrics(f)
+	})
```

Comment:
Could you add a test to test "orphan", similar to this [test](https://github.com/kubernetes/kubernetes/blob/v1.6.0-alpha.0/test/e2e/garbage_collector.go#L170)?